components/public-api/gitpod/v1/organization.proto

```diff
@@ -34,6 +34,20 @@ enum OrganizationRole {
   ORGANIZATION_ROLE_COLLABORATOR = 3;
 }
 
+// OrganizationPermissions define permissions that are restrictable using RoleRestrictions
+enum OrganizationPermission {
+  ORGANIZATION_PERMISSION_UNSPECIFIED = 0;
+  ORGANIZATION_PERMISSION_START_ARBITRARY_REPOS = 1;
+}
+
+message RoleRestrictionEntry {
+  // role is the role that is restricted
+  OrganizationRole role = 1;
+
+  // permissions are the permissions that are restricted
+  repeated OrganizationPermission permissions = 2;
```

Comment:
Will `restriction` better than `allowed`? 
- Like editor and workspace class, we decide to use restriction after a while
- You don't need to migration db rows when add a new restriction. i.e. `CREATE_PROJECT`, once added, you will need to update all db rows?

So how about change to

```
message RolePermissionEntry {
    role = 1;
    repeated OrgPermissionDetail permissions = 2;
}
message OrgPermissionDetail {
    OrgPermission permission = 1;
    bool restricted = 2;
}
enum OrgPermission {
    unspecified = 0;
    // start workspace from any repository
    start_any_repos = 1;
}
```

or

```
message RoleRestrictionEntry {
    role = 1;
    restrictions = 2;
}
enum OrgRestriction {
    unspecified = 0;
    only_start_workspace_from_repo = 1;
}
```

Reply:
The current proto should define just restrictions. When everything is allowed, it will stay empty:

```protobuf
  // permissions are the permissions that are restricted
  repeated OrganizationPermission permissions = 2;
```

Reply:
I see, I thought it's allowed permissions